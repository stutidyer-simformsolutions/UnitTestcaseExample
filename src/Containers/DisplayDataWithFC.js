/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from "react";
import {
  Button,
  FlatList,
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
  View
} from "react-native";

const Section = ({ title, buttonTitle }) => {
  const [data, setData] = React.useState(null);
  const [myArray, setMyArray] = React.useState([]);
  const [isEditPressed, setIsEditPressed] = React.useState(false);
  const [currentIndex, setCurrentIndex] = React.useState(0);
  const onEditPress = (value, index) => {
    setData(value);
    setCurrentIndex(index);
    setIsEditPressed(true);
  };

  const onDeletePress = (index) => {
    setCurrentIndex(index);
    const arr = myArray;
    arr.splice(index, 1);
    setMyArray([...arr]);
  };

  onButtonPress = () => {
    if (isEditPressed) {
      const arr = myArray;
      arr.splice(currentIndex, 1, data);
      setMyArray(arr);
      setIsEditPressed(false);
    } else {
      setMyArray((oldArray) => [...oldArray, data]);
    }
    setData("");
  };
  return (
    <View style={styles.sectionContainer}>
      <Text style={styles.sectionTitle}>{title}</Text>
      <TextInput
        style={styles.input}
        onChangeText={setData}
        value={data}
        placeholder="Please enter the number"
        keyboardType="numeric"
        testID="section-textInput"
      />

      <Button
        disabled={!data ? true : false}
        title={buttonTitle}
        testID="section-button"
        onPress={onButtonPress}
      />
      <DisplayData
        data={myArray}
        onEditPress={onEditPress}
        onDeletePress={onDeletePress}
      />
    </View>
  );
};

const DisplayData = (props) => {
  const { onEditPress, onDeletePress, data } = props;
  const Item = ({ title, index }) => (
    <View style={styles.item}>
      <Text style={styles.title}>{title}</Text>
      <Button
        title={"Edit"}
        onPress={() => onEditPress(title, index)}
        testID={"edit-button"}
      />
      <Button
        title={"Delete"}
        onPress={() => onDeletePress(index)}
        testID={"delete-button"+ index}
      />
    </View>
  );

  const renderItem = ({ item, index }) => {
    return <Item title={item} index={index} />;
  };

  return (
    <FlatList
      data={data}
      testID={'data-flatlist'}
      renderItem={renderItem}
      keyExtractor={(item) => item}
    />
  );
};

const DisplayDataWithFC = () => {
  return (
    <SafeAreaView
      style={styles.mainContainerStyle}
    >
      <Section title={"Enter Value"} buttonTitle={"Press me"} />
    </SafeAreaView>
  );
};

export const styles = StyleSheet.create({
  mainContainerStyle: { flex: 1, marginHorizontal: 25 },
  sectionContainer: {
    marginTop: 32,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: "600",
  },
  item: {
    flexDirection: "row",
    justifyContent: "space-between",
    padding: 5,
  },
  title: {
    fontSize: 17,
    top: 10,
  },
  input: {
    padding: 10,
    height: 40,
    marginVertical: 12,
    borderWidth: 1,
  },
});

export default DisplayDataWithFC;
