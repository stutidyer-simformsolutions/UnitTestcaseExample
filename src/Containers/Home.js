import React from "react";
import { Button, StyleSheet, Text, View } from "react-native";

const Home = (props) => {
  return (
    <View style={styles.container}>
      <Text
        style={styles.textStyle}
      >{`Hello Everyone! \n Welcome to Unit Testcase Demo!`}</Text>
      <View style={styles.buttonContainer}>
        <Button
          title="Demo with Statefull Component"
          onPress={() => props.navigation.navigate("DisplayDataWithClass")}
          style={styles.buttonStyle}
          testID="class-component-button"
        />
        <Button
          title="Demo with Stateless Component"
          onPress={() => props.navigation.navigate("DisplayDataWithFC")}
          style={styles.buttonStyle}
          accessibilityLabel="Press Me"
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  buttonContainer: { flex: 0.8, marginVertical: 25, justifyContent: "center" },
  textStyle: {
    textAlign: "center",
    marginVertical: 8,
    fontSize: 25,
    lineHeight: 50,
  },
  buttonStyle: {
    // height: 120,
    // width: 180
  },
});

export default Home;
