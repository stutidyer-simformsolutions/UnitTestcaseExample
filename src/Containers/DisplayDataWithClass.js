import React, { Component } from "react";
import {
  Button,
  FlatList,
  SafeAreaView,
  Text,
  TextInput,
  View,
} from "react-native";
import { styles } from "./DisplayDataWithFC";

export default class DisplayDataWithClass extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: null,
      myArray: [],
      isEditPressed: false,
      currentIndex: 0,
    };
  }

  onButtonPress = () => {
    const { isEditPressed, myArray, data, currentIndex } = this.state;
    if (isEditPressed) {
      const arr = myArray;
      arr.splice(currentIndex, 1, data);
      this.setState({ myArray: arr, isEditPressed: false });
    } else {
      this.setState((prevState) => ({
        myArray: [...this.state.myArray, this.state.data],
      }));
    }
    this.setState({ data: "" });
  };

  onChangeText = (text) => {
    this.setState({ data: text });
  };

  onEditPress = (item, index) => {
    this.setState({ data: item, currentIndex: index, isEditPressed: true });
  };
  onDeletePress = (index) => {
    const arr = this.state.myArray;
    arr.splice(index, 1);
    this.setState({ myArray: arr });
  };

  renderItem = ({ item, index }) => {
    return (
      <View style={styles.item}>
        <Text style={styles.title}>{item}</Text>
        <Button
          title={"Edit"}
          onPress={() => this.onEditPress(item, index)}
          testID={"edit-button"}
        />
        <Button
          title={"Delete"}
          onPress={() => this.onDeletePress(index)}
          testID={"delete-button"}
        />
      </View>
    );
  };

  renderData = () => {
    const { myArray } = this.state;
    return (
      <FlatList
        contentContainerStyle={{ flexGrow: 1 }}
        data={myArray}
        renderItem={this.renderItem}
        keyExtractor={(item) => item}
      />
    );
  };

  renderContent = () => {
    const { data } = this.state;
    return (
      <View style={styles.sectionContainer}>
        <Text style={styles.sectionTitle}>{"Enter Value"}</Text>
        <TextInput
          style={styles.input}
          onChangeText={this.onChangeText}
          value={data}
          placeholder="Please enter the number"
          keyboardType="numeric"
          testID="section-textInput"
        />

        <Button
          disabled={!data ? true : false}
          title={"Press me"}
          testID="section-button"
          onPress={this.onButtonPress}
        />
        {this.renderData()}
      </View>
    );
  };

  render() {
    return (
      <SafeAreaView style={styles.mainContainerStyle}>
        {this.renderContent()}
      </SafeAreaView>
    );
  }
}
