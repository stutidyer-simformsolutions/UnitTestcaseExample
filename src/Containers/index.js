import Home from "./Home";
import DisplayDataWithClass from "./DisplayDataWithClass";
import DisplayDataWithFC from "./DisplayDataWithFC";

export {DisplayDataWithClass, DisplayDataWithFC, Home};