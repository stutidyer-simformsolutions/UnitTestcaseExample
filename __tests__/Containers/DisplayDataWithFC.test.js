import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import renderer from "react-test-renderer";
import { DisplayDataWithFC } from "../../src/Containers";

describe("Display data with function", () => {
  const { toJSON } = render(<DisplayDataWithFC />);

  // test('Snapshot',() => {
  //     const {toJSON} = render(<DisplayDataWithFC/>)
  //     expect(toJSON()).toMatchSnapshot()
  // })

  test("snapshot of the displayData with functional component", () => {
    const tree = renderer.create(<DisplayDataWithFC />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("when user add any value to the list", async () => {
    const { getByTestId, toJSON } = render(<DisplayDataWithFC />);
    const button = getByTestId("section-button");
    expect(button).toBeDisabled();

    const textInput = await getByTestId("section-textInput");
    fireEvent.changeText(textInput, 10);
    expect(button).toBeEnabled();
    expect(textInput.props.value).toEqual(10);
    fireEvent.press(button);
    expect(toJSON()).toMatchSnapshot();
  });

  test("when user edit any value of list", async () => {
    const { getByTestId } = render(<DisplayDataWithFC />);
    const button = getByTestId("section-button");
    const textInput = await getByTestId("section-textInput");
    fireEvent.changeText(textInput, 20);
    fireEvent.press(button);

    const editButton = await getByTestId("edit-button");
    fireEvent.press(editButton);
    fireEvent.changeText(textInput, 30);
    expect(textInput.props.value).toEqual(30);
    fireEvent.press(button);
  });

  test("when user delete any value from list", async () => {
    const { getByTestId } = render(<DisplayDataWithFC />);
    const button = getByTestId("section-button");

    const textInput = await getByTestId("section-textInput");
    fireEvent.changeText(textInput, 40);
    fireEvent.press(button);

    const deleteButton = await getByTestId("delete-button");
    fireEvent.press(deleteButton);
    expect(textInput.props.value).toEqual("");
  });
});
