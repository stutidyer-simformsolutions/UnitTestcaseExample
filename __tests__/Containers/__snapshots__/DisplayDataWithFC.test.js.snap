// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display data with function snapshot of the displayData with functional component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 25,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Enter Value
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Please enter the number"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "marginVertical": 12,
          "padding": 10,
        }
      }
      testID="section-textInput"
      value={null}
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="section-button"
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="data-flatlist"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`Display data with function when user add value to the list 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 25,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 32,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Enter Value
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Please enter the number"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "marginVertical": 12,
          "padding": 10,
        }
      }
      testID="section-textInput"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="section-button"
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Press me
        </Text>
      </View>
    </View>
    <RCTScrollView
      data={
        Array [
          10,
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="data-flatlist"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 17,
                  "top": 10,
                }
              }
            >
              10
            </Text>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="edit-button"
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Edit
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="delete-button"
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Delete
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
